includes/class-salesforce.php

```diff
@@ -524,13 +625,18 @@ public static function api_sync_salesforce( $request ) {
 		if ( is_array( $orders ) ) {
 			$is_npsp = self::has_field( 'npsp__Primary_Contact__c', 'Opportunity' );
 
-			foreach ( $orders as $order ) {
+			foreach ( $orders as $order_item ) {
 				// Add the NPSP "Primary Contact" field if the Salesforce instance supports it.
 				if ( $is_npsp ) {
-					$order['npsp__Primary_Contact__c'] = $contact_id;
+					$order_item['npsp__Primary_Contact__c'] = $contact_id;
 				}
 
-				$opportunity_response = self::create_opportunity( $order );
+				$existing_opportunity = self::get_opportunity_by_order_id( $order_id, $order_item );
```

Comment:
Calling this method here will make an external API request for every loop iteration. Should we be concerned about performance and rate limits?

Reply:
That's a great point! In my experience, even though each WooCommerce order object has an `orders` array representing all line items for that order, for most Newspack sites every order will only have one line item representing the total donation amount. I think we can revisit this if it becomes a problem. So far it hasn't been an issue for the biggest Newspack site using Salesforce.